sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ResourceEvaluationStage.java

```diff
@@ -63,40 +66,49 @@ public EvaluationOutcome evaluate(
             MesosResourcePool mesosResourcePool,
             OfferRequirement offerRequirement,
             OfferRecommendationSlate offerRecommendationSlate) {
-        ResourceRequirement resourceRequirement = getResourceRequirement();
-        Optional<MesosResource> mesosResourceOptional = mesosResourcePool.consume(resourceRequirement);
-        if (!mesosResourceOptional.isPresent()) {
-            return fail(this, "Failed to satisfy required resource '%s': %s",
-                    resourceRequirement.getName(),
-                    TextFormat.shortDebugString(resourceRequirement.getResource()));
-        }
+        final ResourceRequirement resourceRequirement = getResourceRequirement();
+        final String resourceId = resourceRequirement.getResourceId();
 
-        final MesosResource mesosResource = mesosResourceOptional.get();
-        Resource fulfilledResource = getFulfilledResource(mesosResource);
+        Resource fulfilledResource = getFulfilledResource(new MesosResource(resourceRequirement.getResource()));
         if (resourceRequirement.expectsResource()) {
             logger.info("Expects Resource");
 
+            Optional<MesosResource> existingResourceOptional = mesosResourcePool.getReservedResourceById(resourceId);
```

Comment:
`consume` gives you back what you asked for.  This is checking what exists already, so a determination can be made about whether we need to modify the reservation.  This is the heart of the bug.  For ports it was impossible to know solely from the resource returned by `consume` whether the reservation required modification.

e.g. I want to change from port 5 to 6.  `consume` will return `6`.  That's what I asked for!  Do I need to reserve this?  Unknown based on just the `consume` call, hence the change.